ctapipe/image/muon/muon_processor.py

```diff
@@ -0,0 +1,275 @@
+"""
+High level image processing  (ImageProcessor Component)
+"""
+import numpy as np
+from astropy.coordinates import SkyCoord
+from ctapipe.containers import TelEventIndexContainer
+
+from ctapipe.calib import CameraCalibrator
+from ctapipe.core import TelescopeComponent
+from ctapipe.core import traits
+from ctapipe.io import EventSource
+from ctapipe.image.cleaning import TailcutsImageCleaner
+from ctapipe.coordinates import TelescopeFrame, CameraFrame
+from ctapipe.containers import MuonParametersContainer, ArrayEventContainer
+from ctapipe.instrument import CameraGeometry, SubarrayDescription
+from ctapipe.image.muon import (
+    MuonRingFitter,
+    MuonIntensityFitter,
+    ring_containment,
+    ring_completeness,
+    intensity_ratio_inside_ring,
+    mean_squared_error,
+)
+
+
+class MuonProcessor(TelescopeComponent):
+    """
+    Takes Muon data and cleans and parametrizes the images.
+    """
+
+    completeness_threshold = traits.FloatTelescopeParameter(
+        default_value=30.0, help="Threshold for calculating the ``ring_completeness``"
+    ).tag(config=True)
+
+    ratio_width = traits.FloatTelescopeParameter(
+        default_value=1.5,
+        help=(
+            "Ring width for intensity ratio"
+            " computation as multiple of pixel diameter"
+        ),
+    ).tag(config=True)
+
+    overwrite = traits.Bool(
+        default_value=False, help="If true, overwrite outputfile without asking"
+    ).tag(config=True)
+
+    min_pixels = traits.IntTelescopeParameter(
+        help=(
+            "Minimum number of pixels after cleaning and ring finding"
+            "required to process an event"
+        ),
+        default_value=100,
+    ).tag(config=True)
+
+    pedestal = traits.FloatTelescopeParameter(
+        help="Pedestal noise rms", default_value=1.1
+    ).tag(config=True)
+
+    def __init__(
+        self,
+        subarray: SubarrayDescription,
+        is_simulation,
```

Comment:
There is no need to give this information I think. You can just check `if event.simulation is not None`

Reply:
I think I did that in stage1 since I didn't have the event yet and wanted to set up things

Reply:
As far as I know, this parameter is not used any further in the Processor (and should be recreatable from the EventSource at any time), so I removed it for now.